File: src/main/java/diff/patch/accessors/ListGetter.java
package diff.patch.accessors;

import diff.patch.Getter;

import java.util.List;

/**
 *
 * @author sebasjm <smarchano@primary.com.ar>
 */
public class ListGetter<Element> implements Getter<Element> {
    final int index;
    final List<Element> list;

    public ListGetter(int index, List<Element> list) {
        this.index = index;
        this.list = list;
    }

    @Override
    public Element get() {
        return index < list.size() ? list.get(index) : null;
    }
    
}

File: src/main/java/diff/patch/accessors/ListGetter.kt
package diff.patch.accessors

import diff.patch.Getter

/**

 * @author sebasjm <smarchano></smarchano>@primary.com.ar>
 */
class ListGetter<Element: Any>(internal val index: Int, internal val list: List<Element>) : Getter<Element> {

    override fun get(): Element? {
        return if (index < list.size) list[index] else null
    }

}
